packages/block-library/src/navigation-menu/index.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Server-side rendering of the `core/navigation-menu` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the `core/navigation-menu` block on server.
+ *
+ * @param array $attributes The block attributes.
+ * @param array $content The saved content.
+ * @param array $block The parsed block.
+ *
+ * @return string Returns the post content with the legacy widget added.
+ */
+function render_block_navigation_menu( $attributes, $content, $block ) {
+	return '<nav className="wp-block-navigation-menu">' . build_navigation_menu_html( $block ) . '</nav>';
+}
+
+/**
+ * Walks the inner block structure and returns an HTML list for it.
+ *
+ * @param array $block The block.
+ *
+ * @return string Returns  an HTML list from innerBlocks.
+ */
+function build_navigation_menu_html( $block ) {
+	$html = '';
+	foreach ( (array) $block['innerBlocks'] as $key => $menu_item ) {
```

Comment:
Should we check that `$menu_item` has the expected type of `core/navigation-menu-item`? (We can do this in a follow-up PR.)